src/blockfilter.cpp

```diff
@@ -153,3 +153,64 @@ GCSFilter::GCSFilter(uint64_t siphash_k0, uint64_t siphash_k1, uint8_t P,
 
     bitwriter.Flush();
 }
+
+bool GCSFilter::Match(const GCSFilter::Element& element) const
+{
+    uint64_t query = HashToRange(element);
+
+    CVectorReader stream(GCS_SER_TYPE, GCS_SER_VERSION, m_encoded, 0);
+
+    // Seek forward by size of N
+    uint64_t N = ReadCompactSize(stream);
+    assert(N == m_N);
+
+    BitStreamReader<CVectorReader> bitreader(stream);
+
+    uint64_t value = 0;
+    for (uint64_t i = 0; i < m_N; i++) {
+        uint64_t delta = GolombRiceDecode(bitreader, m_P);
+        value += delta;
+
+        if (query == value) {
+            return true;
+        } else if (query < value) {
+            break;
+        }
+    }
+
+    return false;
+}
+
+bool GCSFilter::MatchAny(const std::set<Element>& elements) const
```

Comment:
In commit "blockfilter: Implement GCSFilter Match methods."

Would suggest implementing `Match` and `MatchAny` in terms of a common

```c++
Match(const uint64_t* sorted_element_hashes, size_t size)
```

method to get rid of all the code duplication between the existing methods. No outside code would need to change, they could just call 

```c++
Match(&query, 1);
Match(queries.data(), queries.size());
```

respectively.

Reply:
Good suggestion.